Mathlib/Analysis/BoxIntegral/UnitPartition.lean

```diff
@@ -293,4 +308,183 @@ theorem prepartition_isPartition {B : Box ι} (hB : hasIntegralVertices B) :
   rw [TaggedPrepartition.mem_toPrepartition, mem_prepartition_iff]
   exact ⟨index n x, mem_admissibleIndex_of_mem_box n hB hx, rfl⟩
 
+theorem exists_admissibleIndex {B : Box ι} (hB : hasIntegralVertices B) {x : ι → ℝ} (hx : x ∈ B) :
+    ∃ ν ∈ admissibleIndex n B, box n ν = box n (index n x) :=
+  ⟨index n x, mem_admissibleIndex_of_mem_box n hB hx, rfl⟩
+
+open Submodule Pointwise BigOperators
+
+open scoped Pointwise
+
+variable (c : ℝ) (s : Set (ι → ℝ)) (F : (ι → ℝ) → ℝ)
+
+local notation "L" => span ℤ (Set.range (Pi.basisFun ℝ ι))
+
+theorem tag_mem_smul_span (ν : ι → ℤ) :
+    tag n ν ∈ (n : ℝ)⁻¹ • L := by
+  rw [ZSpan.smul _ (inv_ne_zero (mod_cast (NeZero.ne n))), Basis.mem_span_iff_repr_mem]
+  refine fun i ↦ ⟨ν i + 1, ?_⟩
+  rw [Basis.repr_isUnitSMul, Pi.basisFun_repr, tag_apply, Units.smul_def, smul_eq_mul,
+    div_eq_inv_mul, ← mul_assoc, IsUnit.val_inv_mul, one_mul, map_add, map_one, eq_intCast]
+
+theorem tag_index_eq_self_of_mem_smul_span {x : ι → ℝ} (hx : x ∈ (n : ℝ)⁻¹ • L) :
+    tag n (index n x) = x := by
+  rw [ZSpan.smul _ (inv_ne_zero (mod_cast (NeZero.ne n))), Basis.mem_span_iff_repr_mem] at hx
+  ext i
+  rsuffices ⟨a, ha⟩ : ∃ a : ℤ, a = n * x i := by
+    specialize hx i
+    rwa [Basis.repr_isUnitSMul, Pi.basisFun_repr, Units.smul_def, Units.val_inv_eq_inv_val,
+      IsUnit.unit_spec, inv_inv] at hx
+  rw [tag_apply, index_apply, Int.cast_sub, Int.cast_one, sub_add_cancel, ← ha, Int.ceil_intCast,
+    ha, mul_div_assoc, mul_div_cancel₀ _ (mod_cast (NeZero.ne n))]
+
+theorem eq_of_index_eq_index_and_mem_smul_span {x y : ι → ℝ} (hx : x ∈ (n : ℝ)⁻¹ • L)
+    (hy : y ∈ (n : ℝ)⁻¹ • L) (h : index n x = index n y) : x = y := by
+  rw [← tag_index_eq_self_of_mem_smul_span n hx, ← tag_index_eq_self_of_mem_smul_span n hy, h]
+
+theorem integralSum_eq_tsum_div {B : Box ι} (hB : hasIntegralVertices B) (hs₀ : s ≤ B) :
+    BoxIntegral.integralSum (Set.indicator s F)
+      (BoxIntegral.BoxAdditiveMap.toSMul (Measure.toBoxAdditive volume))
```

Comment:
```suggestion
    integralSum (Set.indicator s F) (BoxAdditiveMap.toSMul (Measure.toBoxAdditive volume))
```
The namespace `BoxIntegral` is open here. (You can also remove `BoxIntegral.` in the `unfold` line below.)